nixos/modules/services/web-apps/firefly-iii.nix

```diff
@@ -0,0 +1,356 @@
+{ pkgs, config, lib, ... }:
+
+let
+  inherit (lib) optionalString mkDefault mkIf mkOption mkEnableOption literalExpression;
+  inherit (lib.types) nullOr attrsOf oneOf str int bool path package enum submodule;
+  inherit (lib.strings) concatLines removeSuffix hasSuffix;
+  inherit (lib.attrsets) mapAttrsToList filterAttrs;
+  inherit (builtins) isInt isString toString typeOf;
+
+  cfg = config.services.firefly-iii;
+
+  user = cfg.user;
+  group = cfg.group;
+
+  defaultUser = "firefly-iii";
+  defaultGroup = "firefly-iii";
+
+  artisan = "${cfg.package}/artisan";
+
+  env-file-values = filterAttrs (n: v: hasSuffix "_FILE" n) cfg.settings;
+  env-nonfile-values = filterAttrs (n: v: ! hasSuffix "_FILE" n) cfg.settings;
+
+  fileenv-func = ''
+    ${concatLines (mapAttrsToList
+      (n: v: "export ${n}=\"${toString v}\"")
```

Comment:
I'd use `lib.strings.toShellVars` + `lib.concatMapStringsSep "\n" (v: "export ${v}") attrNames`

Reply:
Changed this too. The entire env file is a mess anyway, so changed the approach to use `lib.strings.toShellVar` instead and then just used plain bash loops to do everything else.